miniconf_mqtt/src/lib.rs

```diff
@@ -275,10 +274,20 @@ where
             mqtt: minimq::Minimq::new(stack, clock.clone(), config),
             state: sm::StateMachine::new(sm::Context::new(clock)),
             prefix,
+            alive: "1",
             pending: Multipart::default(),
         })
     }
 
+    /// Set the payload published on the `/alive` topic when connected to the broker.
+    ///
+    /// The default is to publish `1`.
+    /// The message is retained by the broker.
+    /// On disconnect the message is cleared retained through an MQTT will.
+    pub fn set_alive(&mut self, alive: &'buf str) {
+        self.alive = alive;
```

Comment:
Don't we want to put the state machine into a state where the alive message gets republished (in case a user calls this after the alive publication was already sent)

Reply:
I thought about it. But I'd rather not force it.
Let's do a separate fn for reset.